src/component/parser/json/converter/ShapeConverter.ts

```diff
@@ -10,16 +13,38 @@ jsonConvert.ignorePrimitiveChecks = false; // don't allow assigning number to st
 jsonConvert.valueCheckingMode = ValueCheckingMode.DISALLOW_NULL; // never allow null
 //////////////////////////////////////////////////////////////
 
-@JsonConverter
-export class ShapeConverter extends AbstractConverter<Shape[]> {
+abstract class ShapeConverter extends AbstractConverter<Shape[]> {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   deserialize(bpmnDiagram: Array<any> | any): Shape[] {
     try {
       const shapes = bpmnDiagram.BPMNPlane.BPMNShape;
-      return jsonConvert.deserializeArray(ensureIsArray(shapes), Shape);
+      return ensureIsArray(shapes)
+        .filter(shape => this.findShapeElement(shape.bpmnElement))
```

Comment:
not sure it is good idea to call it 2x as it can be costly for big BPMN files

Reply:
talking about .findShapeElement

Reply:
We said the same thing with Thomas, but we don't know to to make the same thing clearly.

Reply:
Ok, as discussed with @csouchet and @aibcmars I am going to restore the if check in the map to avoid double call to the `find` function